### ISSUE TITLE
contourf with X, Y matrix instead of vectors

### ISSUE THREAD

**Issue by sacraig** [matlab2tikz/matlab2tikz#721]
When I call `contourf` using all three coordinate inputs in R2015a, I get a string of errors. The following is a MWE:
```matlab
[X,Y] = meshgrid([1:500],[1:500]);
contourf(X,Y,peaks(500),50,'edgecolor','none')
matlab2tikz('filename','test.tikz')
```
The string of errors are as follows:
```matlab
Error using vertcat
Dimensions of matrices being concatenated are not consistent.

Error in matlab2tikz>drawFilledContours (line 2406)
    zerolevel = [0,          4;

Error in matlab2tikz>drawContourHG2 (line 2299)
      [m2t, str] = drawFilledContours(m2t, str, h, contours, istart, nrows);

Error in matlab2tikz>drawContour (line 2249)
        [m2t, str] = drawContourHG2(m2t, h);

Error in matlab2tikz>drawHggroup (line 2477)
            [m2t,str] = drawContour(m2t, h);

Error in matlab2tikz>handleAllChildren (line 654)
                [m2t, str] = drawHggroup(m2t, child);

Error in matlab2tikz>drawAxes (line 852)
    [m2t, childrenEnvs] = handleAllChildren(m2t, handle);

Error in matlab2tikz>saveToFile (line 450)
        m2t = drawAxes(m2t, relevantAxesHandle);

Error in matlab2tikz (line 352)
saveToFile(m2t, fid, fileWasOpen);
```

This example produces a proper output if the `meshgrid` command and `X` and `Y` parameters within `contourf` are removed, but that is not an option for my purposes.

**Comment by PeterPablo** [comment]
Workaround:
```matlab
[X,Y] = meshgrid([1:500],[1:500]);
contourf(X(1,:),Y(:,1),peaks(500),50,'edgecolor','none')
```

![image](https://cloud.githubusercontent.com/assets/4340267/9191778/886e8d36-4005-11e5-8da0-f3e1ec2dfcb6.png)

There are two issues:
* [ ] m2t does not handle if `X`and `Y` are matrices
* [ ] `edgecolor` is not respected

Thank you for your precise bug report!

**Comment by egeerardyn** [comment]
@PeterPablo: That workaround is okay for many cases, but I think @sacraig 's purpose does not fall within that range.

By specifying the full `X` and `Y` matrices, one can also do some more advanced stuff. E.g.

```
[X,Y] = meshgrid([1:500],[1:500]);
subplot(1,2,1);
contourf(X,Y,peaks(500),50,'edgecolor','none')
subplot(1,2,2);
contourf(X+Y/2,Y,peaks(500),50,'edgecolor','none')
```

<img width="501" alt="screen shot 2015-08-11 at 09 25 57" src="https://cloud.githubusercontent.com/assets/177360/9192395/00ad02b4-400b-11e5-8fcd-289ac66da312.png">

As a first step, we can certainly implement that workaround and maybe even throw a warning if our assumptions about `X` and `Y` are incorrect.

**Comment by PeterPablo** [comment]
You are absolutely correct. I would not implement that workaround in `m2t`. My suggestion should help him continue his work (if he really uses X and Y as shown).

**Comment by egeerardyn** [comment]
I have a patch that might help somewhat. At least, the crashing no longer occurs.
How right the produced output is, however, I cannot guarantee.

At the line where the error occurs (2406 in your case), you could replace the code with
```matlab
zerolevel = [0,          4;
             min(xdata(:)), min(ydata(:));
             min(xdata(:)), max(ydata(:));
             max(xdata(:)), max(ydata(:));
             max(xdata(:)), min(ydata(:))];
```

For my example, I then get the following output.
It looks a lot better, but still not perfect: i.e. the base plane is too wide, but the actual shapes seem to be alright.

<img width="1603" alt="screen shot 2015-08-11 at 10 27 39" src="https://cloud.githubusercontent.com/assets/177360/9193733/ef6fe6e6-4015-11e5-824d-e8f98feaedbf.png">

@sacraig You might want to try out that change in your original use case. For the `meshgrid`, I expect no problems whatsoever to occur anymore.

However, I think that @okomarov might be best suited to have a look at this, since he originally implemented that functionality (if I remember correctly).

**Comment by sacraig** [comment]
Thanks for the rapid response, guys. Thankfully, the first workaround suggested by @PeterPablo is sufficient for my current purposes, as I've been working exclusively on rectangular meshes for this application. I do foresee requiring the non-rectangular grids a bit later down the road on this project, however, so I suspect the workarounds posted by @egeerardyn will be useful as well.

**Comment by egeerardyn** [comment]
Fixed in #722.
